bundles/com.amazonaws.eclipse.cloudformation/src/com/amazonaws/eclipse/explorer/cloudformation/wizard/CreateStackWizardFirstPage1.java

```diff
@@ -0,0 +1,792 @@
+     */
+    private void loadTopics(final Combo snsTopicCombo) {
+        snsTopicCombo.setItems(new String[] { "Loading..." });
+        new Thread() {
+
+            @Override
+            public void run() {
+                AmazonSNS sns = AwsToolkitCore.getClientFactory().getSNSClient();
+                ListTopicsResult topicsResult = sns.listTopics();
+                final List<String> arns = new ArrayList<>();
+                for ( Topic topic : topicsResult.getTopics() ) {
+                    arns.add(topic.getTopicArn());
+                }
+                Display.getDefault().syncExec(new Runnable() {
+
+                    @Override
+                    public void run() {
+                        if ( !snsTopicCombo.isDisposed() ) {
+                            snsTopicCombo.setItems(arns.toArray(new String[arns.size()]));
+                        }
+                    }
+                });
+            }
+
+        }.start();
+    }
+
+    /**
+     * Validates the template file given in a separate thread.
+     */
+    private void validateTemplateFile(String filePath) {
+        CancelableThread.cancelThread(validateTemplateThread);
+        templateValidated.setValue(VALIDATING);
+        try {
+            String fileContents = FileUtils.readFileToString(new File(filePath), "UTF8");
+            validateTemplateThread = new ValidateTemplateThread(
+                    new ValidateTemplateRequest().withTemplateBody(fileContents));
+            validateTemplateThread.start();
+        } catch ( Exception e ) {
+            templateValidated.setValue(INVALID);
+            templateValidationException = e;
+        }
+    }
+
+    /**
+     * Validates the template url given in a separate thread.
+     */
+    private void validateTemplateUrl(String url) {
+        CancelableThread.cancelThread(validateTemplateThread);
+        templateValidated.setValue(VALIDATING);
+        validateTemplateThread = new ValidateTemplateThread(new ValidateTemplateRequest().withTemplateURL(url));
+        validateTemplateThread.start();
+    }
+
+    /**
+     * Cancelable thread to validate a template and update the validation
+     * status.
+     */
+    private final class ValidateTemplateThread extends CancelableThread {
+
+        private final ValidateTemplateRequest rq;
+
+        private ValidateTemplateThread(ValidateTemplateRequest rq) {
+            this.rq = rq;
+        }
+
+        @Override
+        public void run() {
+            ValidateTemplateResult validateTemplateResult;
+            Stack existingStack = null;
+
+            Map templateMap;
+            try {
+
+                // TODO: region should come from context for file-based actions
+                AmazonCloudFormation cf = getCloudFormationClient();
+                validateTemplateResult = cf.validateTemplate(rq);
+
+                if ( wizard.getDataModel().getMode() == Mode.Update && wizard.getDataModel().getStackName() != LOADING_STACKS ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.